model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py

```diff
@@ -691,42 +713,47 @@ def _do_diffusion_step(
             horizontal_start=self._edge_start_lateral_boundary_level_5,
             horizontal_end=self._edge_end_local,
             vertical_start=0,
-            vertical_end=num_levels,
+            vertical_end=self.grid.num_levels,
             offset_provider=self.grid.offset_providers,
         )
         log.debug("running stencils 04 05 06 (apply_diffusion_to_vn): end")
+
         log.debug("communication of prognistic.vn : start")
-        handle_edge_comm = self._exchange.exchange(dims.EdgeDim, prognostic_state.vn)
+        handle_edge_comm = self._exchange(prognostic_state.vn, dim=dims.EdgeDim, wait=False)
 
         log.debug(
             "running stencils 07 08 09 10 (apply_diffusion_to_w_and_compute_horizontal_gradients_for_turbulence): start"
         )
         # TODO (magdalena) get rid of this copying. So far passing an empty buffer instead did not verify?
-        cached.copy_field(prognostic_state.w, self.w_tmp, offset_provider={})
+        cached.copy_field.with_connectivities(self.compile_time_connectivities)(
+            prognostic_state.w, self.w_tmp, offset_provider={}
+        )
 
-        cached.apply_diffusion_to_w_and_compute_horizontal_gradients_for_turbulence(
+        cached.apply_diffusion_to_w_and_compute_horizontal_gradients_for_turbulence.with_connectivities(
+            self.compile_time_connectivities
+        )(
             area=self.cell_params.area,
             geofac_n2s=self.interpolation_state.geofac_n2s,
             geofac_grg_x=self.interpolation_state.geofac_grg_x,
             geofac_grg_y=self.interpolation_state.geofac_grg_y,
             w_old=self.w_tmp,
             w=prognostic_state.w,
-            type_shear=gtx.int32(self.config.shear_type.value),
```

Comment:
that's right. it messes up with the DaCe parser.

Reply:
In case that cannot be fixed easily in dace you could add a reminder in comment that it needs to be like this due to dace. Otherwise somebody might cleanup/change it, breaking dace orchestration...